Content.Shared/ReagentOnItem/Systems/ReagentOnItemSystem.cs

```diff
@@ -0,0 +1,88 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Item;
+using Content.Shared.Popups;
+using Content.Shared.ReagentOnItem;
+using Content.Shared.Fluids;
+using Content.Shared.Clothing.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.FixedPoint;
+using Content.Shared.Chemistry.Reagent;
+
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.ReagentOnItem;
+
+/// <summary>
+///     Deals with items that apply a solution on use to other entities.
+/// </summary>
+public sealed class ReagentOnItemSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPuddleSystem _puddle = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    /// <summary>
+    ///     Tries to apply the given solution to the item.
+    ///     It will apply all reagents that can stick and will spill anything
+    ///     that cannot stick to the floor.
+    /// </summary>
+    /// <returns> False if there is no solution or if entity isn't an item and true otherwise. </returns>
+    public bool ApplyReagentEffectToItem(EntityUid item, Solution reagentMixture)
+    {
+        if (reagentMixture.Volume <= 0 || !HasComp<ItemComponent>(item))
+            return false;
+
```

Comment:
You should use Entity<ItemComponent?> with a resolve instead so if callers have itemcomponent you're avoiding expensive component checks.